tests/test_js_signing.py

```diff
@@ -0,0 +1,55 @@
+"""
+Various tests related to validation for automated signing.
+"""
+
+from js_helper import TestCase
+
+
+class TestSearchService(TestCase):
+    """Tests that warnings related to the search service trigger warnings."""
+
+    def test_changes(self):
+        """Tests that setting changes trigger warnings."""
+
+        def test(obj, prop, stuff, warnings):
+            self.setUp()
+            self.run_script("""%s.%s%s;""" % (obj, prop, stuff))
+            self.assert_failed(with_warnings=warnings)
+
+        objs = ("Cc[''].getService(Ci.nsIBrowserSearchService)",
+                "Services.search")
+
+        for obj in objs:
+            warnings = {"signing_severity": "high",
```

Comment:
On Tue, May 05, 2015 at 07:20:43AM -0700, Mathieu Agopian wrote:
>This can be declared out of the loop, one line above

One of the members gets modified a few lines down. If I moved it 
out of the for loop, I'd have to revert those changes at the 
start of each iteration.